### PR TITLE
[PD-1928] ux for react nuo inbox management

### PR DESCRIPTION
**Pull request by koonsies**
## Overview
This PR represents the UX portion for the non-user outreach inbox management project. The only API contained is modified to return test data rather than actual database contents. This is due to the fact that the add/delete APIs have been shifted to another ticket and it would be cumbersome to create inboxes for testing.

## Testing
To test the UX, ensure you have PRM access by being a company user for a member company (for the old way), or have the following activities 
* "create partner" 
* "create contact"
* "create communication record"

These activities are more or less arbitrary and will be replaced by proper non-user outreach activities as they are developed.

Navigate to {root}/prm/view/nonuseroutreach

# Future Work
* Add/Edit API
* Delete API
* URL Routing (need input from Daniel)
* Unit Tests for API
* Unit Tests for JS
* Creation of activities and proper view wrapping
* UI modifications based on activities

### CODE REVIEW

**Comment by lemniscate613** [comment]
More later, but... the gap to the right of the form might be eliminated (unless it's being used somehow).

I'll help you fold in all of my UI work Monday (Respond.js for IE8, routing, responsive styles).

**Comment by aspidites** [comment]
re: folding new work - Does said work dramatically change what @koonsies has here, or is it just icing on the cake? I'd be worried about enlarging this PR by introducing even more unevaluated work by a second developer.

**Comment by aspidites** [comment]
Haven't started reviewing this in detail, but my immediate concern is a lack of unit tests. In addition to there being new django views which we've historically written tests for, we now (thanks to @darrint ) have a javascript testing framework which I'd like us to start using. Especially considering the amount that we will seemingly be relying on for current and future projects.

**Comment by lemniscate613** [comment]
@aspidites Good point. That other work will be apart of a different PR.

**Comment by koonsies** [comment]
There will definitely be unit tests for this work. As I have not yet done JS unit testing, I am spinning that off into another ticket with the Python unit tests.

File: gulp/gulpfile.js [hunk 1]
```diff
@@ -102,11 +102,40 @@ gulp.task('manageusers', function() {
...
+    })
+    .on('package', function(pkg) {
+        util.log("Including package:", pkg.name)
+    })
+    .pipe(source('nonuseroutreach.js'))
+    .pipe(buffer())
+    .pipe(sourcemaps.init({loadMaps: true}))
+    // Do we want this in production builds?
+    .pipe(uglify({ mangle: false }))
+    // stripDebug() must come before sourcemaps.write()
```
Comment (lemniscate613) [inline gulp/gulpfile.js]: These lines can be removed. We're not doing this anymore.

File: gulp/gulpfile.js [hunk 2]
```diff
@@ -102,11 +102,40 @@ gulp.task('manageusers', function() {
...
+    // You should remove logging before committing, but this confirms logging won't be in production
+    .pipe(gulpif(strip_debug, stripDebug()))
+    .pipe(sourcemaps.write('./'))
+    .pipe(gulp.dest(dest))
+});
+
 // By default, we strip logging. This disables that functionality.
 gulp.task('watch-no-strip', function() {
```
Comment (lemniscate613) [inline gulp/gulpfile.js]: This will be gone in a future branch.

File: gulp/src/nonuseroutreach/nonuseroutreach.js [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+    );
+  }
+});
+
+// validation method to ensure value is a proper email username
+// returns:
+// Object
+//    -success (whether or not the object validated)
+//    -message (error message from validator)
+var validateEmailInput = function(value){
```
Comment (lemniscate613) [inline gulp/src/nonuseroutreach/nonuseroutreach.js]: I've broken basic email validation out into an exported function in /util/. We could use it in a future PR and still keep your '@' check if you want.

File: gulp/src/nonuseroutreach/nonuseroutreach.js [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+// the "master" container
+var Container = React.createClass({
+  getInitialState: function() {
+    return {
+      page:"Overview",
+      company: "DirectEmployers",
+      tips: [],
+    };
+  },
+  // change what it displayed in the "Content" div. option parameter "tips" allows a list of tips to display
```
Comment (lemniscate613) [inline gulp/src/nonuseroutreach/nonuseroutreach.js]: Interesting idea. I haven't been taking advantage of the "tips" area...

File: templates/nonuseroutreach/nuo_main.html [hunk 5]
```diff
@@ -0,0 +1,63 @@
+{% extends "base-new.html" %}
+
+{% block site-title %}
+    <title>My.jobs - {{ company.name }} - Non-User Outreach Inbox Management</title>
+    <meta name="title" content="My.jobs - Non-User Outreach Inbox Management">
+{% endblock %}
+
+{% block meta-extra %}
+<!-- <link rel="stylesheet" href="{{STATIC_URL}}pickadate/themes/classic.css" id="theme_base">
```
Comment (lemniscate613) [inline templates/nonuseroutreach/nuo_main.html]: All commented lines in this file can removed. It's all a remnant of the prototype I started with.

File: gulp/src/nonuseroutreach/nonuseroutreach.js [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+      return <span></span>;
+    }
+    return (
+        <div className="col-xs-12 ">
+          <div className="wrapper-header">
+            <h2>Existing Inbox Management</h2>
+          </div>
+          <div className="partner-holder">
+        {inboxes.map((inbox_ob, i) =>
+              <InboxRow inbox={inbox_ob} key={inbox_ob.pk} index={i} handleDelete={this.handleDelete} loadInboxesFromApi={this.loadInboxesFromApi} />
```
Comment (lemniscate613) [inline gulp/src/nonuseroutreach/nonuseroutreach.js]: You are passing in a unique key here...

File: gulp/src/nonuseroutreach/nonuseroutreach.js [hunk 7]
```diff
@@ -0,0 +1,455 @@
...
+      buttons: [new ControlButton("Delete", false, true)],
+      validation_messages: []
+    }
+  },
+  render: function(){
+    var validation_messages = this.state.validation_messages.map((message) =>
+    <HelpText message={message} />
+    );
+    return (
+    <div className="clearfix product-card no-highlight">
```
Comment (lemniscate613) [inline gulp/src/nonuseroutreach/nonuseroutreach.js]: You should be able to grab that unique key you passed in, here. Then we'll get rid of that error with the "Inbox Management" view loads, "Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `Menu`. See https://fb.me/react-warning-keys for more information."

Comment (koonsies) [inline gulp/src/nonuseroutreach/nonuseroutreach.js]: corrected

File: gulp/src/nonuseroutreach/nonuseroutreach.js (hunk 3 above)
Comment (koonsies) [inline gulp/src/nonuseroutreach/nonuseroutreach.js]: addressed

File: gulp/src/nonuseroutreach/nonuseroutreach.js (hunk 6 above)
Comment (koonsies) [inline gulp/src/nonuseroutreach/nonuseroutreach.js]: addressed

**Comment by aspidites** [comment]
Not a fan of python tests being spun off into a separate ticket. As for the js ones, I could argue that there are already tests for @darrint's stuff, so we should lean on him as a resource in getting those tests integrated into PRs.

**Comment by koonsies** [comment]
Closed to add unit tests

**Pull request closed without merging**